pygeoapi/linked_data.py

```diff
@@ -173,104 +175,146 @@ def jsonldify_collection(cls, collection, locale_):
                     config['server']['url'], dataset, feature['id'])
 
-            if not geojsonld:
-                feature, geocontext = make_jsonld(feature)
-                geocontext.append({
-                    "features": "schema:itemListElement",
-                    "FeatureCollection": "schema:itemList"
-                })
-                # Note: @id or https://schema.org/url, both or something else?
-                feature[id_field] = identifier
-            else:
-                feature['id'] = identifier
-
-            data['features'][i] = feature
+            # Note: @id or https://schema.org/url, both or something else?
+            data['features'][i] = {
+                id_field: identifier,
+                'type': 'schema:Place'
+            }
 
     if data.get('timeStamp', False):
         data['https://schema.org/sdDatePublished'] = data.pop('timeStamp')
 
-    defaultVocabulary = "https://geojson.org/geojson-ld/geojson-context.jsonld"
-    if not geojsonld:
-        ldjsonData = {
-            "@context": [
-                {
-                 "schema": "https://schema.org/",
-                 id_field: "@id",
-                 "type": "@type",
-                },
-                *(context or []),
-                *(geocontext or [])
-            ],
-            **data
-        }
-    else:
-        ldjsonData = {
-            "@context": [
-                defaultVocabulary,
-                *(context or [])
-            ],
-            **data
-        }
+    data['links'] = data.pop('links')
+
+    ldjsonData = {
+        '@context': [defaultVocabulary, *(context or [])],
+        **data
+    }
 
     return ldjsonData
 
 
-def make_jsonld(feature):
-    # Expand properties block
-    feature = {**feature, **feature.get('properties')}
-    feature.pop('properties')
+def jsonldify_geometry(feature):
+    """
+        Render JSON-LD for feature with GeoJSON, Geosparql/WKT, and
+        schema geometry encodings.
+
+        :param feature: feature body to with GeoJSON geometry
+
+        :returns: None
+    """
+
+    geo = feature.get('geometry')
+    geom = asShape(geo)
+
+    # GeoJSON geometry
+    feature['geometry'] = feature.pop('geometry')
+
+    # Geosparql geometry
+    feature['geosparql:hasGeometry'] = {
+        '@type': f'http://www.opengis.net/ont/sf#{geom.geom_type}',
+        'geosparql:asWKT': {
+            '@type': 'http://www.opengis.net/ont/geosparql#wktLiteral',
+            '@value': f'{geom.wkt}'
+        }
+    }
 
-    feature['type'] = 'schema:Place'
+    # Schema geometry
+    feature['schema:geo'] = {'@type': 'schema:GeoShape'}
+    if geom.geom_type == 'Point':
+        feature['schema:geo'] = {
+            '@type': 'schema:GeoCoordinates',
+            'schema:longitude': geom.x,
+            'schema:latitude': geom.y
+        }
+        return
+
+    elif geom.geom_type == 'LineString':
+        _ = [f'{x},{y}' for (x, y) in geom.coords[:]]
```

Comment:
I wonder if these operations can be split out into smaller dedicated functions, that can then also be tested in isolation.

Reply:
I think there is a lot of space here for improvement of this section. I don't even know the name of this coordinate system. It resembles the geometry section of GML but there doesn't appear to be an easy way to transform to these shapes. I left it all in the same function because I was trying to keep `geom` isolated to within jsonldify geometry. The ideal situation would probably be an addition to shapely geometry outputs written in C/cython. Interested to hear what you think is the best short and long term implementations would look like.

Reply:
For now we've isolated the geometry to schema.org conversion function, but not gone so far as to make a function for each of the geom types for the medium term. Schema.org geometry is so nonstandard and hard to parse I doubt anything useful could get into main shapely branch anytime soon though.